packages/java-edition/src/mcfunction/parser/argument.ts

```diff
@@ -1767,3 +1822,187 @@ function vector(
 		return ans
 	}
 }
+
+const components: core.InfallibleParser<ComponentListNode> = core.map(
+	core.record({
+		start: '[',
+		pair: {
+			key: core.resourceLocation({ category: 'data_component_type' }),
+			sep: '=',
+			value: nbt.parser.entry,
+			end: ',',
+			trailingEnd: true,
+		},
+		end: ']',
+	}),
+	(res) => {
+		const ans: ComponentListNode = {
+			type: 'mcfunction:component_list',
+			range: res.range,
+			children: res.children,
+		}
+		return ans
+	},
+)
+
+const componentTest: core.Parser<ComponentTestNode> = (src, ctx) => {
+	const start = src.cursor
+	src.skipWhitespace()
+	const negated = src.trySkip('!')
+	src.skipWhitespace()
+
+	const resLoc = core.resourceLocation({ category: 'data_component_type' })(
+		src,
+		ctx,
+	)
+	src.skipWhitespace()
+
+	if (src.trySkip('=')) {
+		const value = nbt.parser.entry(src, ctx)
+
+		if (value == core.Failure) {
+			ctx.err.report(localize('expected', localize('nbt.node')), src)
+			src.skipUntilOrEnd(',', '|', ']')
+			return core.Failure
```

Comment:
This should not return a failure. Since you already reported an error, it should return a `ComponenttestExactNode` but with `value` undefined.

Reply:
See 5cfdfa1fb9b8d9c393409a2c1252e7ad42f7b458